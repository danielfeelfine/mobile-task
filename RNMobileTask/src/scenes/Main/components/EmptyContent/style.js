import { StyleSheet } from 'react-native'
import { colors } from '@styles/base'

const style = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 38,
    paddingHorizontal: 16,
    alignItems: 'center',
    backgroundColor: colors.Light,
  },
  title: {
    textAlign: 'center',
    fontWeight: '800',
    fontSize: 18,
    lineHeight: 22,
    color: colors.Gray,
  },
  message: {
    marginVertical: 12,
    paddingHorizontal: 12,
    textAlign: 'center',
    fontWeight: '300',
    fontSize: 12,
    lineHeight: 18,
    color: colors.Gray,
  },
})

export default style
