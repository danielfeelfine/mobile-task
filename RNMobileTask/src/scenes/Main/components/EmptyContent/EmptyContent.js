// @flow
import React from 'react'
import { Text, View } from 'react-native'
import style from './style'

const EmptyContent = () => {
  return (
    <View style={style.container}>
      <Text style={style.title}>There are no content to show</Text>
      <Text style={style.message}>Sorry! We don't have content to show for this news source. Please choose another one.</Text>
    </View>
  )
}

export default EmptyContent
