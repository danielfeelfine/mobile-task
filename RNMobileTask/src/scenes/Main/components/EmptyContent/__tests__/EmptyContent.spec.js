import React from 'react'
import { shallow } from 'enzyme'
import EmptyContent from './../EmptyContent'


describe('EmptyContent component', () => {

  it('renders correctly', () => {
    const wrapper = shallow(
      <EmptyContent />
    )
  
    expect(wrapper).toMatchSnapshot()
  })

})
