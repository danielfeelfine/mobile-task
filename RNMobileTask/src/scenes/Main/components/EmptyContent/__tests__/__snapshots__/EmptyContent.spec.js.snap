// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyContent component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyContent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 18,
              "fontWeight": "800",
              "lineHeight": 22,
              "textAlign": "center",
            }
          }
        >
          There are no content to show
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 18,
              "marginVertical": 12,
              "paddingHorizontal": 12,
              "textAlign": "center",
            }
          }
        >
          Sorry! We don't have content to show for this news source. Please choose another one.
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 38,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "There are no content to show",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#999999",
            "fontSize": 18,
            "fontWeight": "800",
            "lineHeight": 22,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "There are no content to show",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Sorry! We don't have content to show for this news source. Please choose another one.",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#999999",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 18,
            "marginVertical": 12,
            "paddingHorizontal": 12,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Sorry! We don't have content to show for this news source. Please choose another one.",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontSize": 18,
                "fontWeight": "800",
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            There are no content to show
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontSize": 12,
                "fontWeight": "300",
                "lineHeight": 18,
                "marginVertical": 12,
                "paddingHorizontal": 12,
                "textAlign": "center",
              }
            }
          >
            Sorry! We don't have content to show for this news source. Please choose another one.
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 38,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "There are no content to show",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#999999",
              "fontSize": 18,
              "fontWeight": "800",
              "lineHeight": 22,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "There are no content to show",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Sorry! We don't have content to show for this news source. Please choose another one.",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 18,
              "marginVertical": 12,
              "paddingHorizontal": 12,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Sorry! We don't have content to show for this news source. Please choose another one.",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
